src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -475,7 +475,8 @@ def _save_checkpoint_shard(
             # convert the state
             if isinstance(obj, Module) and isinstance(obj, XLAFSDP):
                 converted = obj.state_dict()
-                # add shard_metadata to state
+                # add shard_metadata to state. this format is defined by
+                # https://github.com/pytorch/xla/blob/v2.1.0/torch_xla/distributed/fsdp/state_dict_utils.py#L122-L125
```

Comment:
I wonder what happens if you save a state where the key for the model is not "model" 

```
fabric.save(path, {"banana": model})
```
This would be totally valid in any other setting, but I think here it would fail since the XLA format expects these keys.

Reply:
iirc this won't work when trying to consolidate the checkpoint using `consolidate_sharded_model_checkpoints` https://github.com/Lightning-AI/lightning/pull/18746/files/e5b695aff64bc37ae1a67fba4aac4981200eecfd#diff-3908a573abf00ae5f37061f214f2a3c2616b6591e0c96206b9f48b4c7ab49ea4R457. I'm not sure how this works for individual shards.